eip7251: Bugfix and more withdrawal tests (#14578)
prysmaticlabs/prysm: CHANGELOG.md

- Fixed mesh size by appending `gParams.Dhi = gossipSubDhi`
- Fix skipping partial withdrawals count.
- wait for the async StreamEvent writer to exit before leaving the http handler, avoiding race condition panics [pr](https://github.com/prysmaticlabs/prysm/pull/14557)

> - EIP7521 - fixing withdrawal bug

Reviewer: Can we be more specific with this change log? High level: what is the bug?

Reviewer 2: It's a really minor detail, but shouldn't it be 7251 instead of 7521?

> - EIP7521 - fixing withdrawal bug, pending partial withdrawals was not taken account of in the withdrawal sweep. The already withdrawn balance should be deducted from the balance considered in the sweep.

Reviewer 3: Sentence nit
```suggestion
- EIP7521 - Fixes withdrawal bug by accounting for pending partial withdrawals and deducting already withdrawn amounts from the sweep balance.
```
